clap-utils/src/keypair.rs

```diff
@@ -142,27 +143,35 @@ pub(crate) enum SignerSource {
     Pubkey(Pubkey),
 }
 
-pub(crate) fn parse_signer_source<S: AsRef<str>>(source: S) -> SignerSource {
+#[derive(Debug, Error)]
+pub(crate) enum SignerSourceError {
+    #[error("unrecognized signer source")]
+    UnrecognizedSource,
+}
+
+pub(crate) fn parse_signer_source<S: AsRef<str>>(
+    source: S,
+) -> Result<SignerSource, SignerSourceError> {
     let source = source.as_ref();
     match uriparse::URIReference::try_from(source) {
-        Err(_) => SignerSource::Filepath(source.to_string()),
+        Err(_) => Err(SignerSourceError::UnrecognizedSource),
         Ok(uri) => {
             if let Some(scheme) = uri.scheme() {
                 let scheme = scheme.as_str().to_ascii_lowercase();
                 match scheme.as_str() {
-                    "ask" => SignerSource::Ask,
-                    "file" => SignerSource::Filepath(uri.path().to_string()),
-                    "stdin" => SignerSource::Stdin,
-                    "usb" => SignerSource::Usb(source.to_string()),
-                    _ => SignerSource::Filepath(source.to_string()),
+                    "ask" => Ok(SignerSource::Ask),
+                    "file" => Ok(SignerSource::Filepath(uri.path().to_string())),
```

Comment:
wdyt about `stat()`ing the path here and failing if it doesn't exist with readable permissions?

Reply:
Great idea

Reply:
Done in: https://github.com/solana-labs/solana/pull/16933/commits/b354deaac92d6ce72be6fe14f24dff7d1dff8fa8
Wdyt?